datacube_stats/main.py

```diff
@@ -674,13 +678,22 @@ def __call__(self, index, sources_spec, date_ranges):
             for source_spec in sources_spec:
                 group_by_name = source_spec.get('group_by', DEFAULT_GROUP_BY)
                 source_filter = source_spec.get('source_filter', None)
-                data = workflow.list_cells(product=source_spec['product'], time=time_period,
-                                           group_by=group_by_name, geopolygon=self.geopolygon,
-                                           cell_index=self.cell_index, source_filter=source_filter)
-                masks = [workflow.list_cells(product=mask['product'], time=time_period,
-                                             group_by=group_by_name, geopolygon=self.geopolygon,
-                                             cell_index=self.cell_index)
-                         for mask in source_spec.get('masks', [])]
+
+                all_products = [source_spec['product']] + [m['product'] for m in source_spec.get('masks', [])]
+
+                product_query = {all_products[0]: {'source_filter': source_filter}}
+
+                all_data, unmatched_ = multi_product_list_cells(all_products, workflow,
```

Comment:
`Stats` is its own codebase, it’s a great place to go Python 3 only.

We need to start the move for everything else too...